server/application/application.go

```diff
@@ -309,15 +300,69 @@ func (s *Server) List(ctx context.Context, q *application.ApplicationQuery) (*v1
 		return newItems[i].Name < newItems[j].Name
 	})
 
+	if q.Offset != nil && q.Limit != nil {
+		if *q.Offset < 0 || *q.Limit < 0 {
+			return nil, fmt.Errorf("offset %d and limit %d must be a non-negative integer", *q.Offset, *q.Limit)
+		}
+		if *q.Offset >= int64(len(newItems)) || *q.Limit == 0 {
+			newItems = make([]v1alpha1.Application, 0)
+		} else {
+			if *q.Offset+*q.Limit >= int64(len(newItems)) {
+				newItems = newItems[*q.Offset:]
+			} else {
+				newItems = newItems[*q.Offset : *q.Offset+*q.Limit]
+			}
+		}
+	}
+
 	appList := v1alpha1.ApplicationList{
```

Comment:
It seems that supporting pagination in the Watch API is one of the proposal's goals.

![image](https://github.com/user-attachments/assets/4a423546-7731-40b1-bad6-785207e4c8ac)

Reply:
ok I will add it